ui/contexts/metamask-notifications/metamask-notifications.tsx

```diff
@@ -38,6 +45,18 @@ export const MetamaskNotificationsProvider: React.FC = ({ children }) => {
   const { listNotifications, notificationsData, isLoading, error } =
     useListNotifications();
 
+  const wasProfileSyncingDisabled =
+    isProfileSyncingEnabledRef.current && !isProfileSyncingEnabled;
+
+  useEffect(() => {
+    if (wasProfileSyncingDisabled) {
+      // list notifications to update the counter
+      listNotifications();
```

Comment:
Yep good that you call this! As snap notifications should still be added the counter (which this will do).

If you want, I'm happy to get this into the notification controller as an event to listen to?
```ts
// inside notification controller
.subscribe(
  '...',
  (newVal, oldVal) => {
    if (oldVal === true && newVal === false) {
      this.listNotifications() // internal call to update list.
    }
  }
)
```

Reply:
OK fair, lets see if it needs to be done in mobile first!